### PR TITLE
Lexical Editor

### PR DESCRIPTION
**Pull request by andreistangacianu**

### CODE REVIEW

File: stories/lexical/index.stories.css
Comment (antalutrazvan) [inline stories/lexical/index.stories.css:1]: Remove this file

File: stories/lexical/index.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+<Story 
+  name="Text Editor" 
+  parameters={{
+    layout: "centered",
+  }}
+>
+  {() => {
+      const [text, setText] = useState("");
+      return <Editor onChange={() => {}} value="test default value" />;
```
Comment (antalutrazvan) [inline stories/lexical/index.stories.mdx:46]: @andreistangacianu why is the story contents commented out ?

File: stories/lexical/index.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  }}
+</Story>
+
+The component can be imported from:
+
+```jsx
+import Editor from "@figshare/fcl/textEditor";
+
+class MyTextEditor extends React.Component {
```
Comment (antalutrazvan) [inline stories/lexical/index.stories.mdx:55]: @andreistangacianu @alinacoca I would suggest adding more content to the story itself, and to this `jsx` code showcase, to show how `toolbar-items` can be configured as well.

File: stories/lexical/index.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+## Examples
+
+<Canvas>
+  <div style={canvasStyle}>
+    <p>Basic - empty</p>
+    <Editor
+      className={styles.editorWrapper}
+      placeholder="Type a text..."
+      onChange={() => {}}
```
Comment (antalutrazvan) [inline stories/lexical/index.stories.mdx:77]: @andreistangacianu add an entry with a custom toolbar as well. static code only.

File: packages/ui/textEditor/Lexical/plugins/LinkEditor/useModal.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        onClose={onClose}
+      >
+        {content}
+      </Overlay>
+    );
+  }, [modalContent, onClose]);
+
+  const showModal = useCallback(
+    (
+      title,
```
Comment (antalutrazvan) [inline packages/ui/textEditor/Lexical/plugins/LinkEditor/useModal.jsx:37]: @alinacoca Interesting approach. using a `hook` to mount a component and it's content 🤔 

I would suggest the argument for this function be an `object`:
`useCallback({ title, getContent, isVisible }) => { ... }, [onClose, setModalContent])` - not sure what `isVisible` is used for. will there be instances where isVisible is false ?

File: packages/ui/textEditor/Lexical/plugins/LinkEditor/LinkEditor.test.jsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
+import Enzyme, { mount } from "enzyme";
+import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
+import React from "react";
+
+import OpenLinkIcon from "./components/OpenLinkIcon";
+import UnlinkIcon from "./components/UnlinkIcon";
+import { LinkEditor } from "./LinkEditor";
+
+
+Enzyme.configure({ adapter: new Adapter() });
```
[the remark this answers is not in the document]
Comment (andreistangacianu) [reply packages/ui/textEditor/Lexical/plugins/LinkEditor/LinkEditor.test.jsx:9]: I'm unsure why that's added there - cc @alinacoca

**Review by antalutrazvan — APPROVED** [review]

**Pull request merged**